openwisp_radius/api/serializers.py

```diff
@@ -126,10 +132,39 @@ def get_pdf_link(self, obj):
     def validate(self, data):
         if data['strategy'] == 'prefix' and not data.get('number_of_users'):
             raise serializers.ValidationError(
-                {'number_of_users': 'The field number_of_users cannot be empty'}
+                {'number_of_users': 'The field number_of_users cannot be empty.'}
+            )
+        try:
+            self.batch_org = Organization.objects.get(slug=data['organization_slug'])
+        except Organization.DoesNotExist:
+            raise serializers.ValidationError(
+                {'organization_slug': 'Organization not found.'}
+            )
+        try:
+            RadiusBatch.objects.get(name=data['name'], organization=self.batch_org)
+        except RadiusBatch.DoesNotExist:
+            pass
+        else:
+            raise serializers.ValidationError(
+                {'name': 'RadiusBatch with same name exists for organization.'}
             )
         return super().validate(data)
 
+    def create(self, validated_data):
+        strategy = validated_data.get('strategy')
+        options = dict(
+            name=validated_data.get('name'),
+            strategy=strategy,
+            expiration_date=validated_data.get('expiration_date'),
+            organization=self.batch_org,
+        )
+        if strategy == 'csv':
+            options.update(csvfile=validated_data.get('csvfile'))
+            return RadiusBatch.objects.create(**options)
+        elif strategy == 'prefix':
+            options.update(prefix=validated_data.get('prefix'))
+            return RadiusBatch.objects.create(**options)
```

Comment:
This could be:
```python
if strategy == 'csv':
    options.update(csvfile=validated_data.get('csvfile'))
elif strategy == 'prefix':
    options.update(prefix=validated_data.get('prefix'))
return RadiusBatch.objects.create(**options)
```

But why wouldn't the default create method work out of the box without the need of being overridden?
What is the mandatory step that has to be performed?
Adding the org? is the if really necessary?